ArduPlane/takeoff.cpp

```diff
@@ -305,3 +305,29 @@ void Plane::landing_gear_update(void)
     g2.landing_gear.update(relative_ground_altitude(g.rangefinder_landing));
 }
 #endif
+
+/*
+ check takeoff_timeout; checks time after the takeoff start time; returns true if timeout has occurred and disarms on timeout
+*/
+
+bool Plane::check_takeoff_timeout(void)
+{
+    if (takeoff_state.start_time_ms != 0 && g2.takeoff_timeout > 0) {
```

Comment:
This method will look a *lot* better with doing early-returns rather than testing

Reply:
It would, but I don't think its a requirement for this.